apps/opik-backend/src/main/java/com/comet/opik/domain/llmproviders/OpenAi.java

```diff
@@ -0,0 +1,92 @@
+package com.comet.opik.domain.llmproviders;
+
+import com.comet.opik.infrastructure.LlmProviderClientConfig;
+import dev.ai4j.openai4j.OpenAiClient;
+import dev.ai4j.openai4j.OpenAiHttpException;
+import dev.ai4j.openai4j.chat.ChatCompletionRequest;
+import dev.ai4j.openai4j.chat.ChatCompletionResponse;
+import jakarta.inject.Inject;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+@Slf4j
+public class OpenAi implements LlmProviderService {
+    private final LlmProviderClientConfig llmProviderClientConfig;
+    private final OpenAiClient openAiClient;
+
+    @Inject
+    public OpenAi(LlmProviderClientConfig llmProviderClientConfig, String apiKey) {
+        this.llmProviderClientConfig = llmProviderClientConfig;
+        this.openAiClient = newOpenAiClient(apiKey);
+    }
+
+    @Override
+    public ChatCompletionResponse generate(@NonNull ChatCompletionRequest request, @NonNull String workspaceId) {
+        return openAiClient.chatCompletion(request).execute();
+    }
+
+    @Override
+    public void generateStream(
+            @NonNull ChatCompletionRequest request,
+            @NonNull String workspaceId,
+            @NonNull Consumer<ChatCompletionResponse> handleMessage,
+            @NonNull Runnable handleClose,
+            @NonNull Consumer<Throwable> handleError) {
+        openAiClient.chatCompletion(request)
+                .onPartialResponse(handleMessage)
+                .onComplete(handleClose)
+                .onError(handleError)
+                .execute();
+    }
+
+    @Override
+    public void validateRequest(ChatCompletionRequest request) {
+
+    }
+
+    @Override
+    public Optional<LlmProviderError> getLlmProviderError(Throwable runtimeException) {
+        if (runtimeException instanceof OpenAiHttpException openAiHttpException) {
+            return Optional.of(LlmProviderError.builder()
+                    .code(openAiHttpException.code())
+                    .message(openAiHttpException.getMessage())
+                    .build());
+        }
+
+        return Optional.empty();
+    }
+
+    /**
+     * At the moment, openai4j client and also langchain4j wrappers, don't support dynamic API keys. That can imply
+     * an important performance penalty for next phases. The following options should be evaluated:
+     * - Cache clients, but can be unsafe.
+     * - Find and evaluate other clients.
+     * - Implement our own client.
+     * TODO as part of : <a href="https://comet-ml.atlassian.net/browse/OPIK-522">OPIK-522</a>
+     */
+    private OpenAiClient newOpenAiClient(String apiKey) {
```

Comment:
This could be moved to a module and encapsulated by exposing some builder component, which can even cache the client if necessary.

Reply:
Agreed to fix as part of #987